setup/base/base.go

```diff
@@ -205,6 +205,7 @@ func NewBaseDendrite(cfg *config.Dendrite, componentName string, options ...Base
 		PublicKeyAPIMux:        mux.NewRouter().SkipClean(true).PathPrefix(httputil.PublicKeyPathPrefix).Subrouter().UseEncodedPath(),
 		PublicMediaAPIMux:      mux.NewRouter().SkipClean(true).PathPrefix(httputil.PublicMediaPathPrefix).Subrouter().UseEncodedPath(),
 		PublicWellKnownAPIMux:  mux.NewRouter().SkipClean(true).PathPrefix(httputil.PublicWellKnownPrefix).Subrouter().UseEncodedPath(),
+		PublicConsentAPIMux:    mux.NewRouter().SkipClean(true).PathPrefix("/_matrix").Subrouter().UseEncodedPath(),
```

Comment:
This feels just a bit off — it's really a mux for the whole `/_matrix` namespace. Are we actually required to use `/_matrix/consent` specifically as the endpoint?

Reply:
Since it's not in the spec, we are free to decide where it serves. I just followed what Synapse did. :P

Reply:
Maybe it's OK for it to live in the `PublicClientAPIMux` as `/_matrix/client/consent` then, and we don't have to pass around a router for the entirety of `/_matrix`, since that seems like a potential footgun waiting to happen.